ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -523,6 +523,8 @@ protected Boolean findNextCompactionAndExecute(boolean computeStats) throws Inte
       final StatsUpdater su = computeStats ? StatsUpdater.init(ci, msc.findColumnsWithStats(
           CompactionInfo.compactionInfoToStruct(ci)), conf,
           runJobAsSelf(ci.runAs) ? ci.runAs : t.getOwner()) : null;
+      Path resultDir = QueryCompactor.Util.getCompactionResultDir(sd, tblValidWriteIds,
```

Comment:
Can you plz double check this, as the params can be diff depending on major/min/MM etc.